rotkehlchen/exchanges/coinbase.py

```diff
@@ -160,21 +161,117 @@ def __init__(
+        - 'nbf': The "not before" timestamp, set to the current time.
+        - 'exp': The expiration timestamp, set to 2 minutes from the current time.
+        - 'uri': The provided URI.
+
+        The token is signed using the ES256 algorithm and includes a unique 'nonce' in the headers.
+
+        Args:
+            uri (str): The URI for which the JWT token is being generated.
+
+        Returns:
+            str: The generated JWT token.
+
+        Raises:
+            RemoteError: If there is an error during the JWT token generation process.
+        """
+        try:
+            private_key = serialization.load_pem_private_key(self.secret, password=None)
+
+            current_time = int(time.time())
+            jwt_payload = {
+                'sub': self.api_key,
+                'iss': "coinbase-cloud",
+                'nbf': current_time,
+                'exp': current_time + 120,
+                'uri': uri,
+            }
+            jwt_token = jwt.encode(
+                jwt_payload,
+                private_key,
+                algorithm='ES256',
+                headers={'kid': self.api_key, 'nonce': secrets.token_hex()},
+            )
+            return jwt_token
+        except (jwt.PyJWTError, ValueError) as e:
+            raise RemoteError(f'Error generating JWT token: {str(e)}') from e
+        
+    def validate_api_key(self) -> tuple[bool, str]:
+        """
+        Validates that the Coinbase API key is good for usage in rotki.
+        Makes sure that the following permissions are given to the key:
+        wallet:accounts:read, wallet:transactions:read, wallet:withdrawals:read, wallet:deposits:read
+        """
+        if self.is_legacy_api_key:
+            # Validate legacy API key
+            result, msg = self._validate_single_api_key_action('accounts')
+            if result is None:
+                return False, msg
+
+            # now get the account ids
+            account_info = self._get_active_account_info(result)
+            if len(account_info) != 0:
+                # and now try to get all transactions of an account to see if that's possible
+                method = f'accounts/{account_info[0][0]}/transactions'
+                result, msg = self._validate_single_api_key_action(method)
+                if result is None:
+                    return False, msg
 
+            return True, ''
+        else:
+            # Validate new API key format
+            api_key_pattern = r'^organizations/[\w-]+/apiKeys/[\w-]+$'
+            if not re.match(api_key_pattern, self.api_key):
+                return False, 'Invalid Coinbase API key name format'
+
+            private_key_pattern = r'^-----BEGIN EC PRIVATE KEY-----\n[\w+/=\n]+-----END EC PRIVATE KEY-----\n?$'
+            if not re.match(private_key_pattern, self.secret.decode('utf-8'), re.MULTILINE):
+                return False, 'Invalid Coinbase private key format'
+            return True, ''
```

Comment:
readability for last return

```suggestion

            return True, ''
```